torch/csrc/distributed/c10d/init.cpp

```diff
@@ -292,9 +292,40 @@ void _register_builtin_comm_hook(
   reducer.register_builtin_comm_hook(comm_hook_type);
 }
 
+std::atomic<bool> event_queue_enabled = false;
+int sync_pipe;
+std::mutex event_queue_lock;
+std::deque<::c10d::EventInfo> event_queue;
+
+void enqueue_event_for_python(const ::c10d::EventInfo& evt) {
+  if (!event_queue_enabled.load())
+    return;
+
+  std::unique_lock<std::mutex> lock(event_queue_lock);
+  event_queue.push_back(evt);
+  char m = 'x';
+  write(sync_pipe, &m, 1);
+}
+
+void enable_event_collection(int pipe) {
```

Comment:
I think this should be renamed to `enable_python_event_collection' or something that has python in the name.  I had to double check, but i believe python/c++ callbacks are truly independent as they should be.  (in other words, you don't have to call this enabler to make your c++ callbacks work, and the c++ callbacks are in no way sharing infrastructure with the python one.  So the `event_queue_enabled` and `event_queue` vars should probably also have python in their name.